### ISSUE TITLE
CA1859 is not reported when concrete type is an array

### ISSUE THREAD

**Issue by drieseng** [dotnet/roslyn-analyzers#6565]
### Analyzer

**Diagnostic ID**: [CA1859]: `Change return type to concrete type for improved performance'

### Analyzer source

**NuGet Package**: [Microsoft.CodeAnalysis.NetAnalyzers](https://www.nuget.org/packages/Microsoft.CodeAnalysis.NetAnalyzers)

**Version**: 8.0.0-preview1.23121.1

### Describe the bug

**CA1859** is not reported when the concrete return type is an array.

### Steps To Reproduce

**For example:**

Use the following code fragment:

```c#
using System;
using System.Collections.Generic;

public class X
{
    private static IList<string> GetListPrivate()
    {
        return Array.Empty<string>();
    }
}
```

and the following **.editorconfig**:

```
[*.cs]
dotnet_diagnostic.CA1859.severity = error
```

### Expected behavior

The **CA1859** diagnostic is reported for the `GetListPrivate()` method proposing to change the return type from **IList\<string>** to **string[]**.

### Actual behavior

**CA1859** is not reported.
